torch_trainer.py

```diff
@@ -126,13 +126,20 @@ def train(self):
         """
         assert self.trainer is not None, "Please make sure the trainer is successfully initialized by `self._setup_trainer()`."
         train_loader = self._get_dataset_loader(split='train', shuffle=self.config.shuffle)
-        val_loader = self._get_dataset_loader(split='val')
-        self.trainer.fit(self.model, train_loader, val_loader)
 
-        # Set model to current best model
+        if 'val' not in self.datasets:
+            logging.info('No validation dataset is provided. Train without vaildation.')
+            self.trainer.fit(self.model, train_loader)
+        else:
+            val_loader = self._get_dataset_loader(split='val')
+            self.trainer.fit(self.model, train_loader, val_loader)
+
+        # Set model to the best model. If the validation process is skipped during
+        # training (i.e., val_size=0), the model is set to the last model.
         checkpoint_callback = [callback for callback in self.trainer.callbacks if isinstance(callback, ModelCheckpoint)][0]
-        logging.info(f'Finished training. Load best model from {checkpoint_callback.best_model_path}')
-        self._setup_model(checkpoint_path=checkpoint_callback.best_model_path)
+        model_path = checkpoint_callback.best_model_path if checkpoint_callback.best_model_path else checkpoint_callback.last_model_path
```

Comment:
```suggestion
        model_path = checkpoint_callback.best_model_path or checkpoint_callback.last_model_path
```
This is a more pythonic way to write `A if A else B`.